echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/artifacts/ArtifactMatcher.java

```diff
@@ -86,6 +93,55 @@ public static boolean isConstraintInPayload(final Map constraints, final Map pay
     return true;
   }
 
+  /**
+   * Check that there is a key in the payload for each constraint declared in a Trigger. Also check
+   * that if there is a value for a given key, that the value matches the value in the payload.
+   *
+   * <p>The constraint key may accept a JsonPath expression for deeper json search, the evaluation
+   * value should be a String or a List<String>.
+   *
+   * @param constraints A map of constraints configured in the Trigger (eg, created in Deck). A
+   *     constraint is a [key, java regex value] pair or a [JsonPathExp, java regex value].
+   * @param payload A map of the payload contents POST'd in the triggering event.
+   * @return Whether every key or expression (and value if applicable) in the constraints map is
+   *     represented in the payload.
+   */
+  public static boolean isJsonPathConstraintInPayload(final Map constraints, final Map payload) {
+    String json = gson.toJson(payload);
+    DocumentContext documentContext = JsonPath.using(conf).parse(json);
+
+    for (Object key : constraints.keySet()) {
+      if (!payload.containsKey(key) || payload.get(key) == null) {
+        String value;
+        try {
+          value = documentContext.read(key.toString(), String.class);
+        } catch (InvalidPathException e) {
+          return false;
+        }
+        if (value != null) {
+          return matches(constraints.get(key).toString(), value);
```

Comment:
What happens if you have two or more jsonpath constraints? It looks like you'll return `true` early here without checking the second constraint.

Reply:
This is my only real concern. If what I've written isn't true I would still add a test that demonstrates that it works with multiple constraints.

Reply:
thanks for taking a look, this does not work when multiple constraints are evaluated. I pushed the changes that will support multiple constraints and a test to demonstrate it.